web_programming/emails_from_url.py

```diff
@@ -1,103 +1,96 @@
-"""Get the site emails from URL."""
-__author__ = "Muhammad Umer Farooq"
-__license__ = "MIT"
-__version__ = "1.0.0"
-__maintainer__ = "Muhammad Umer Farooq"
-__email__ = "contact@muhammadumerfarooq.me"
-__status__ = "Alpha"
+"""
+Author - AyushSehrawat
+"""
 
 import re
-from html.parser import HTMLParser
-from urllib import parse
+import urllib.parse
+from collections import deque
 
 import requests
+import requests.exceptions
 
+# Imports the needed modules
+from bs4 import BeautifulSoup
 
-class Parser(HTMLParser):
-    def __init__(self, domain: str):
-        HTMLParser.__init__(self)
-        self.data = []
-        self.domain = domain
-
-    def handle_starttag(self, tag: str, attrs: str) -> None:
-        """
-        This function parse html to take takes url from tags
-        """
-        # Only parse the 'anchor' tag.
-        if tag == "a":
-            # Check the list of defined attributes.
-            for name, value in attrs:
-                # If href is defined, and not empty nor # print it.
-                if name == "href" and value != "#" and value != "":
-                    # If not already in data.
-                    if value not in self.data:
-                        url = parse.urljoin(self.domain, value)
-                        self.data.append(url)
-
-
-# Get main domain name (example.com)
-def get_domain_name(url: str) -> str:
-    """
-    This function get the main domain name
-
-    >>> get_domain_name("https://a.b.c.d/e/f?g=h,i=j#k")
-    'c.d'
-    >>> get_domain_name("Not a URL!")
-    ''
-    """
-    return ".".join(get_sub_domain_name(url).split(".")[-2:])
+""" Function to get emails from the url """
 
 
-# Get sub domain name (sub.example.com)
-def get_sub_domain_name(url: str) -> str:
+def getMails(num: int) -> None:
+    count = 0
     """
-    >>> get_sub_domain_name("https://a.b.c.d/e/f?g=h,i=j#k")
-    'a.b.c.d'
-    >>> get_sub_domain_name("Not a URL!")
-    ''
```

Comment:
Why are we REMOVING tests?!?

Reply:
Can you elaborate more? You mean a test case?